Content.Shared/Chemistry/Components/HyposprayComponent.cs

```diff
@@ -0,0 +1,49 @@
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+using Robust.Shared.Audio;
+
+namespace Content.Shared.Chemistry.Components;
+
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class HyposprayComponent : Component
+{
+    [DataField]
+    public string SolutionName = "hypospray";
+
+    // TODO: This should be on clumsycomponent.
+    [DataField]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public float ClumsyFailChance = 0.5f;
+
+    [DataField]
+    [ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 TransferAmount = FixedPoint2.New(5);
+
+    [DataField]
+    public SoundSpecifier InjectSound = new SoundPathSpecifier("/Audio/Items/hypospray.ogg");
+
+    /// <summary>
+    /// Whether or not the hypo is able to inject only into mobs. On false you can inject into beakers/jugs
+    /// </summary>
+    [AutoNetworkedField]
+    [DataField]
+    public HyposprayToggleMode ToggleMode = HyposprayToggleMode.All;
```

Comment:
This could be a boolean instead of an enum, if it's only checking if it's allowed to inject into mobs or not. Although it's a good idea to think about possible future functionality

Reply:
It used to be a boolean before, but I switched over to an enum to describe what the "false" behavior would be (All targets injectable). I don't mind switching back to a boolean.